runtime/compiler/optimizer/UnsafeFastPath.cpp

```diff
@@ -142,16 +154,24 @@ void TR_UnsafeFastPath::transformUnsafeAtomicCallInVarHandleAccessMethod(TR::Tre
       {
       // codegen doesn't optimize CAS on a static field
       //
-      if (callerMethod != TR::java_lang_invoke_StaticFieldVarHandle_StaticFieldVarHandleOperations_OpMethod)
+      if (callerMethod != TR::java_lang_invoke_StaticFieldVarHandle_StaticFieldVarHandleOperations_OpMethod &&
```

Comment:
This seems error prone. How are you certain there won't be new VarHandle methods introduced which operate on static fields? If this happens (and it's certainly not unlikely) this check would pass and we would perform an unsafe operation.

Rather than opt-out for statics, can we instead opt-in for non-statics? i.e. turn the `!=` into a set of `==`? Is this feasible? Do you understand my concern?

Reply:
I don't think that is feasible since there are a large number of other VarHandle sub-types. The static case is specific so I think this is reasonable unless @liqunl has some specific idea on how to group 'everything else'.

Reply:
There're not many VarHandle sub-types. The thing is that whoever adds a sub-type has to let the JIT know, otherwise the jitted code will crash. So I think either way is fine.